omc3/scripts/kmod_summary.py

```diff
@@ -0,0 +1,221 @@
+from omc3.scripts.create_logbook_entry import main
+from omc3.utils import logging_tools
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+from generic_parser import DotDict
+
+LOG = logging_tools.get_logger(__name__)
+
+def output_kmod_summary_tables(
+        meas_paths: Sequence[Path | str],
+        averaged_meas: dict[str, dict[int, tfs.TfsDataFrame]] = None,
+        output_dir: Path | str | None = None,
+        logbook: str = None
+        ) -> tuple[dict[str, list[tfs.TfsDataFrame]],dict[str, list[list[str]]]]:
+
+    """
+    Args:
+        meas_paths (Sequence[Path | str]): Paths to the K-modulation results.
+        averaged_meas (dict[int, tfs.TfsDataFrame]): If not None, averaged K-modulation results over all measurements are included in the summary tables. Default: None.
+        output_dir (Path | str | None): Path to the output directory. Defaults to None.
+        logbook (str = None): If provided, create a logbook entry containing the .txt summary tables to the given logbook. Default: None.
+
+    Returns:
+        Tuple[dict[str, tfs.TfsDataFrame], dict[str, list[str]]]:
+            - Dictionary mapping beam names to K-modulation summary DataFrames.
+            - Dictionary mapping beam names to lists of formatted text tables
+    """
+    LOG.info("Starting kmod summary importing.")
+    cols_x = [f"{BETASTAR}X", f"{ERR}{BETASTAR}X", f"{BETAWAIST}X", f"{ERR}{BETAWAIST}X", f"{WAIST}X", f"{ERR}{WAIST}X"]
+    cols_y = [f"{BETASTAR}Y", f"{ERR}{BETASTAR}Y", f"{BETAWAIST}Y", f"{ERR}{BETAWAIST}Y", f"{WAIST}Y", f"{ERR}{WAIST}Y"]
+    beams = [f"{BEAM_DIR}{1}", f"{BEAM_DIR}{2}"]
+
+    grouped = {beam: [] for beam in beams}
+    """
+    Gathers the various Kmod results.tfs dataframes, taking only cols_x and cols_y values.
+    Then it stores them in grouped, divided by beam.
+    """
+    for path in meas_paths:
+        path = Path(path)
+        LOG.info(f"Reading results: {path}.")
+        for beam in beams:
+            file_path = path / beam / "results.tfs"
+            if not file_path.exists():
+                LOG.warning(f"Missing results.tfs: {file_path}")
+            else:
+                file_name = file_path.parent.parent.name
+                data_file = tfs.read(file_path)
+                try:
+                    label = data_file[f"{LABEL}"].iloc[0]
+                except KeyError as e:
+                    LOG.debug("Could not parse IP from LABEL. Skipping entry.", exc_info=e)
+                    ip_name = "None"
+                else:
+                    second_magnet = label.split("-")[1]
+                    relevant = second_magnet.split(".")[-1]
+                    ip_number = relevant[1:]
+                    ip_name = f"IP{ip_number}"
+                df = data_file[cols_x + cols_y].iloc[[0]]
+                df.insert(0, "NAME", file_name)
+                df.insert(0, "IP", ip_name)
+                grouped[beam].append(df)
+
+    grouped_averaged = {beam: [] for beam in beams}
```

Comment:
Should this go below the if statement of next line? I guess there's no point in creating this variable if we don't have `averaged_meas` provided.